File: geeksforgeeks/21.2.cpp
#include <bits/stdc++.h>
using namespace std;

#define int long long

bool can_sort(int arr[], int n){
    vector <int> a, b;
    for(int i = 0; i < n; i++){
        a.push_back(arr[i]);
    }
    b = a;
    if(n <= 3){
        return true;
    }
    sort(b.begin(), b.end());
    vector <int> d(b.rbegin(), b.rend());
    for(int i = 0; i < n-1; i++){
        for(int j = i+1; j < n; j++){
            vector <int> c = a;
            reverse(c.begin()+i, c.begin()+j+1);
            if(c == b || c == d){
                return true;
            }
        }
    }
    return false;
}

int32_t main(){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int tt;
    cin >> tt;
    while(tt--){
        int n;
        cin >> n;
        int arr[n];
        for(int i = 0; i < n; i++) cin >> arr[i];
        if(can_sort(arr, n)){
            cout << "YES" << endl;
        }
        else{
            cout << "NO" << endl;
        }
    }
    
}

File: geeksforgeeks/21-2.py
def can_sort(arr, n):
    A = [x for x in arr]
    A.sort()
    if len(A) <= 3:
        return True
    for i in range(n-1):
        for j in range(i+1, n):
            newarr = []
            if i > 0:
                newarr.extend(arr[:i])
            newarr.extend(reversed(arr[i:j+1]))
            if j < n-1:
                newarr.extend(arr[j+1:])

            if newarr == A or newarr == A[::-1]:
                return True
    return False

n = int(input())
arr = [int(s) for s in input().split()]
if can_sort(arr, n) == True:
    print("YES")
else:
    print("NO")


        
